src/backend/core/tests/test_api_users.py

```diff
@@ -32,8 +34,144 @@ def test_api_users_list_authenticated():
     response = APIClient().get(
         "/api/v1.0/users/", HTTP_AUTHORIZATION=f"Bearer {jwt_token}"
     )
-    assert response.status_code == 404
-    assert "Not Found" in response.content.decode("utf-8")
+    assert response.status_code == 200
+    assert len(response.json()) == 3
+
+
+def test_api_users_authenticated_list_by_email():
+    """
+    Authenticated users should be able to search users with a case insensitive and
+    partial query on the email.
+    """
+    identity = factories.IdentityFactory(email="tester@ministry.fr")
```

Comment:
I'm setting the email for the authenticated user because tests sometimes fail because of random emails containing queried strings.